src/components/Layout/Nav.tsx

```diff
@@ -83,14 +83,19 @@ const Nav = memo(({ onClick }: NavProps) => {
   const state = useGlobalState();
 
   const [isPlaying, setIsPlaying] = useState(false);
+  const [links, setLinks] = useState<LinkItem[]>([]);
 
   const handleMouseEnter = () => {
     setIsPlaying(true);
   };
 
+  useEffect(() => {
+    createLinks(state.tenantName).then((links) => setLinks(links));
```

Comment:
I assume that the creation of the links, which entails a call to the backend, would usually be quite fast so that the menu will be shown shortly after the app launch.

However, could it be that it takes longer than a second or more? In that case maybe we can show the menu right from the beginning and either a) leave out the "Buy PEN" entry and only show once the link has loaded or b) show the "Buy PEN" entry right away but disable it as long as the link is not available.

Reply:
Thanks for the comment. I changed it so that the function is not async anymore but returns a 'default' array of LinkItems and a promise that resolves to contain the actual fully-prepared LinkItems.

Reply:
I am not sure whether I am reading this right: the [initial link](https://github.com/pendulum-chain/portal/pull/543/files/baaee22c3adad97d21cf4c2d12d3284a950d6429..200787c43a223149b15fde6f2b3ae630b84ab68e#diff-0156eb8485cbc4ab310a91b31d411940c81a71b430bcc84a0974a800765a94c6R145) will be `config.alchemyPay.prodUrl` and will then be replaced by the correct link once the promise resolved. Does that initial link (which is defined to be `https://ramp.alchemypay.org`) make sense to click on?

Reply:
Yes, the only difference between `https://ramp.alchemypay.org` and the link we generate is that on the proper link, you will see 'PEN' prefilled in the field for the receiving currency. The rest is the same.

Reply:
Perfect. Thanks for the change.